pkg/runtime/reconciler.go

```diff
@@ -211,9 +214,13 @@ func (r *resourceReconciler) Sync(
 	isAdopted := IsAdopted(desired)
 	rlog.WithValues("is_adopted", isAdopted)
 
-	// TODO(jaypipes): Validate all dependent resources. The AWSResource
-	// interface needs to get some methods that return schema relationships,
-	// first though
+	rlog.Enter("rm.ResolveReferences")
+	resolvedRefDesired, err := rm.ResolveReferences(ctx, r.apiReader, desired)
+	rlog.Exit("rm.ResolveReferences", err)
```

Comment:
Would it be better to delegate the logging to the generated `ResolveReferences` here?

Reply:
This is consistent with other resourceManager invocations from reconciler.go. 
This provides the nice resource logger step (">>") for any logs printed from rm.ResolveReferences